### DISCUSSION TITLE
Mermaid's stability

### DISCUSSION THREAD

**Discussion by Sun-ZhenXing** [vuepress-theme-hope/vuepress-theme-hope#2820] [Q&A] [answered]
There have been some issues with using Mermaid@10.0.0, and these should be fixed later or upstream. (beta179/beta180 have been tested).

The current test does not work: `does not provide an export named`, point to `mermaid -> moment`.

------------
PS: Problem that occurred before Mermaid@10 that I encountered: **Incompatibility with containers**.

```markdown
::: details

<!-- Mermaid here -->

:::
```

When details hide, all the arrows of the Mermaid containers on the page are missing. And flicker when first rendered. Any other ideas?

**Comment by Mister-Hope** [comment]
You can see that official documentation and ci worked. So you should provide reproduction about this.

**Reply by Sun-ZhenXing** [reply]
```bash
pnpm create vuepress-theme-hope .

? Select a language to display / 选择显示语言 简体中文
? 选择包管理器 pnpm
生成 package.json...
? 设置应用名称 vuepress-theme-hope-template
? 设置应用版本号 2.0.0
? 设置应用描述 A project of vuepress-theme-hope
? 设置协议 MIT
? 是否需要一个自动部署文档到 GitHub Pages 的工作流？ No
? 你想要创建什么类型的项目？ docs
生成模板...
? 是否初始化 Git 仓库? Yes

devDependencies:
+ @vuepress/client 2.0.0-beta.60
+ vue 3.2.47
+ vuepress 2.0.0-beta.60
+ vuepress-theme-hope 2.0.0-beta.181
```

Add mermaid code to `README.md`:

```md
\```mermaid
graph TD
  1
  2
  3
\```
```

Dev:

```bash
pnpm docs:dev
```

console:

```log
config-5161385b.js?v=2d511ae7:2 
        
       Uncaught (in promise) SyntaxError: The requested module '/@fs/D:/workspace/VuePress/test/node_modules/.pnpm/moment-mini@2.29.4/node_modules/moment-mini/moment.min.js?v=2d511ae7' does not provide an export named 'default' (at config-5161385b.js?v=2d511ae7:2:8)
```

Mermaid keeps loading.

-----
By the way, when I am developing in dev mode, my page content is lost when `vite` print `[page reload] ...`, is there a good solution? This could be something else.

**Comment by Mister-Hope** [comment, marked as answer]
Should be fixed, available in next version.

**Reply by Sun-ZhenXing** [reply]
All repaired, thanks.

**Reply by Mister-Hope** [reply]
HI, we revert this change in https://github.com/vuepress-theme-hope/vuepress-theme-hope/commit/08b2ec220c1bf9574c61112ea1f792ceb5c02526 because mermaid10.0.1 now use dayjs, you may be aware of this change in new version.
